app/src/main/kotlin/com/hippo/ehviewer/gallery/MetaPageLoader.kt

```diff
@@ -0,0 +1,129 @@
+package com.hippo.ehviewer.gallery
+
+import androidx.collection.SieveCache
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.image.Image
+import com.hippo.ehviewer.util.OSUtils
+import com.hippo.ehviewer.util.isAtLeastO
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+
+private const val MAX_CACHE_SIZE = 512 * 1024 * 1024
+private const val MIN_CACHE_SIZE = 128 * 1024 * 1024
+
+abstract class MetaPageLoader : AutoCloseable {
```

Comment:
What's the benefit of keeping this instead of merging with `PageLoader`?

Reply:
IDK, I will merge if you like it

Reply:
Done